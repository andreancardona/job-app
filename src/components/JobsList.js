import React from 'react';
import { useHistory } from 'react-router-dom';
import CurrentDate from './CurrentDate';
import '../styles/JobsList.css';

// JobList component where pass down props: jobs & editJob
// Here we will handle the list view for all the jobs
const JobsList = (props) => {

  let history = useHistory();
  
  return (
    <div className="jobs-list-container">
      <div className="jobs-list-headers" tabIndex="0">
        <div className="header-title">Job Title</div>
        <div className="header-posted">Posted</div>
        <div className="header-sponsorship">Sponsorship</div>
        <div className="header-status">Status</div>
      </div>
      {/* We are mapping over our list of jobs to be rendered and viewed. 
      If data was not already provided check for items in list  */}
      {props.jobs.map((job) => (
        <div className="job-tile" key={job.id}>
          <div className="job-title-location">
            <div className="title">{job.title}</div>
            <div className="location">{job.location}</div>
          </div>
            <div className="posted"><CurrentDate /></div>
            <div className="sponsorship">{job.sponsorship}</div>
            <div className="status">{job.status}</div>
          <button className="edit-button" onClick={() => {props.editJob(job) || history.push("/edit-job")}}>
            Edit
          </button>
          <div>
          </div>
        </div>
      ))}
    </div>
  )
}

export default JobsList;