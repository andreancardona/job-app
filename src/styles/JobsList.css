/* jobs list container */
.jobs-list-container {
  width: 70%;
  border-radius: 4px;
  margin: 50px 0 50px 0;
}


@media screen and (max-width: 768px) {
  .jobs-list-container {
    width: 95%;
    margin: 15% 0 10% 0;
  };
}

.jobs-list-headers {
  font-family: "Noto Sans";
  font-size: 14px;
  font-style: normal;
  font-weight: 700;
  line-height: 21px;
  letter-spacing: 0em;
  text-align: left;
  color:  #949494;
  padding: 10px 0 10px 0;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

@media screen and (max-width: 768px) {
  .jobs-list-headers {
    font-family: Noto Sans;
    font-size: 16px;

    display: flex;
    flex-direction: row;
    justify-content: flex-start;
  }
}

.header-title {
  padding: 0 100px 0 10px;
}

.header-status {
  padding-right: 150px;
}

@media screen and (max-width: 768px) {
  .header-title {
    padding: 0 8% 0 3%;
  }

  .header-posted {
    padding-right: 10px;
  }

  .header-sponsorship {
    padding-right: 10px;
  }

  .header-status {
    padding-right: 10px;
  }
}

.job-tile {
  align-items: center;
  background: #fff;
  display: flex;
  font-size: 20px;
  justify-content: space-between;
  margin-bottom: 6px;

  width: 100%;
  height: 80px;
  border: 1px solid #D4D2D0;
  box-sizing: border-box;
  border-radius: 8px;
}

.job-title-location {
  width: 200px;
  padding: 18px 0px 18px 18px;
}

.title {
  font-family: "Noto Sans";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 21px;
  letter-spacing: 0em;
  text-align: left;
}

.location {
  font-family: "Noto Sans";
  font-size: 14px;
  font-style: normal;
  font-weight: 400;
  line-height: 21px;
  letter-spacing: 0em;
  text-align: left;
  color: #949494;
}

@media screen and (max-width: 600px) {
  .job-tile {
    font-size: 16px;
  }

  .job-title-location {
    width: 23%;
    padding: 5% 0px 5% 5%;
  }
}

.edit-button {
  margin: 0 20px 0 0;
}

@media screen and (max-width: 600px) {
  .edit-button {
    margin: 0px;
  }
}