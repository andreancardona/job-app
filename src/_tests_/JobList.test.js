import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import JobsList from '../components/JobsList';

// Initial list of jobs
const jobs = [
  { id: 1, title: 'Product Manager', location: 'Austin, TX', posted: '10/04/2020', sponsorship: 'Free', status: 'Open' },
  { id: 2, title: 'CEO', location: 'Austin, TX', posted: '12/30/2020', sponsorship: 'Sponsored', status: 'Paused'},
  { id: 3, title: 'Software Engineer', location: 'Seattle, WA', posted: '11/18/2020', sponsorship: 'Free', status: 'Closed' },
]

describe('The <JobsList /> component', () => {
  test('JobsList renders with JobsList props', () => {
    const tree = renderer.create(<JobsList jobs={jobs}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('JobsList should not fail any accessibility tests', async () => {
    const { container } = render(<JobsList jobs={jobs}/>);
    expect(await axe(container)).toHaveNoViolations();
  })
});
