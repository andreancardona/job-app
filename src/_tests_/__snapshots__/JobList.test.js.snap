// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <JobsList /> component JobsList renders with JobsList props 1`] = `
<div
  aria-hidden="true"
  className="jobs-list-container"
  id="job-list"
  role="grid"
>
  <div
    aria-label="jobs listings"
    className="jobs-list-headers"
    id="job-list-headers"
    role="row"
    tabIndex="0"
  >
    <div
      aria-label="title"
      className="header-title"
    >
      Job Title
    </div>
    <div
      aria-label="posted"
      className="header-posted"
    >
      Posted
    </div>
    <div
      aria-label="sponsorhsip"
      className="header-sponsorship"
    >
      Sponsorship
    </div>
    <div
      aria-label="status"
      className="header-status"
    >
      Status
    </div>
  </div>
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleProduct Manager"
        className="title"
      >
        Product Manager
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted10/04/2020"
        className="posted"
      >
        10/04/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusOpen"
        className="status"
      >
        Open
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleCEO"
        className="title"
      >
        CEO
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted12/30/2020"
        className="posted"
      >
        12/30/2020
      </div>
      <div
        aria-label="sponsorshipSponsored"
        className="sponsorship"
      >
        Sponsored
      </div>
      <div
        aria-label="statusPaused"
        className="status"
      >
        Paused
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleSoftware Engineer"
        className="title"
      >
        Software Engineer
      </div>
      <div
        aria-label="locationSeattle, WA"
        className="location"
      >
        Seattle, WA
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted11/18/2020"
        className="posted"
      >
        11/18/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusClosed"
        className="status"
      >
        Closed
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
